src/main/java/ch/njol/skript/expressions/ExprName.java

```diff
@@ -97,336 +96,151 @@
-					} else {
-						if (name == null)
-							copy = Bukkit.createInventory(inventory.getHolder(), invType);
-						else
-							copy = Bukkit.createInventory(inventory.getHolder(), invType, name);
-					}
-					copy.setContents(inventory.getContents());
-					viewers.forEach(human -> human.openInventory(copy));
-				} else {
-					assert false;
-				}
-			}
-
-			@Override
-			@Nullable
-			String get(@Nullable Object o) {
-				if (o == null)
-					return null;
-				if (o instanceof Player) {
-					return ((Player) o).getDisplayName();
-				} else if (o instanceof Entity) {
-					return ((Entity) o).getCustomName();
-				} else if (o instanceof ItemStack) {
-					if (!((ItemStack) o).hasItemMeta())
-						return null;
-					ItemMeta m = ((ItemStack) o).getItemMeta();
-					return m == null || !m.hasDisplayName() ? null : m.getDisplayName();
-				} else if (o instanceof Inventory) {
-					if (TITLE_METHOD != null) {
-						try {
-							return ((String) TITLE_METHOD.invoke(o));
-						} catch (IllegalAccessException e) {
-							assert false;
-							return null;
-						} catch (Throwable e) {
-							Skript.exception(e);
-							return null;
-						}
-					}
-					return null;
-				} else {
-					assert false;
-					return null;
-				}
-			}
-		},
-		TABLIST_NAME("player list name", "(player|tab)[ ]list name[s]", PLAYER, PLAYER) {
-			@Override
-			void set(@Nullable Object o, @Nullable String name) {
-				if (o == null)
-					return;
-				if (o instanceof Player) {
-					try {
-						((Player) o).setPlayerListName(name == null ? "" : name);
-					} catch (IllegalArgumentException ignored) {}
-				} else {
-					assert false;
-				}
-			}
-
-			@Override
-			@Nullable
-			String get(@Nullable Object o) {
-				if (o == null)
-					return null;
-				if (o instanceof Player) {
-					return ((Player) o).getPlayerListName();
-				} else {
-					assert false;
-					return null;
-				}
-			}
-		};
-
-		final String name;
-		final String pattern;
-		final int from;
-		final int acceptChange;
-
-		NameType(String name, String pattern, int from, int change) {
-			this.name = name;
-			this.pattern = "(" + ordinal() + "¦)" + pattern;
-			this.from = from;
-			acceptChange = change;
-		}
-
-		abstract void set(@Nullable Object o, @Nullable String s);
-
-		@Nullable
-		abstract String get(@Nullable Object o);
-
-		String getFrom() {
-			StringBuilder b = new StringBuilder();
-			for (int i = 0; i < types.length; i++) {
-				if ((from & (1 << i)) == 0)
-					continue;
-				if ((1 << i) == ITEM && !Skript.isRunningMinecraft(1, 4, 5))
-					continue;
-				if ((1 << i) == ENTITY && !Skript.isRunningMinecraft(1, 5))
-					continue;
-				if (b.length() != 0)
-					b.append("/");
-				b.append(types[i]);
-			}
-			return "" + b;
-		}
-	}
-
-	static {
-		for (NameType n : NameType.values())
-			register(ExprName.class, String.class, n.pattern, n.getFrom());
-	}
-
-	@SuppressWarnings("null")
-	private NameType type;
+	/**
+	 * 1 = "name",

```

Comment:
This implementation detail might be better as a normal multi-line comment rather than a javadoc.